### PR TITLE
BLOCKCHAIN-176 Added proposal fee and updated tests

### PR DESCRIPTION
**Pull request by MichalLustykNeti**

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -265,7 +265,7 @@ impl frame_system::Config for Runtime {
 
 #[cfg(not(feature = "testnet-runtime"))]
 parameter_types! {
-	pub const LaunchPeriod: BlockNumber = 36 * DAYS;
+	pub const LaunchPeriod: BlockNumber = 10;
```
Comment (kacperzuk-neti) [inline bin/node/runtime/src/lib.rs:268]: Revert this.

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -792,6 +792,7 @@ parameter_types! {
 	pub const EnactmentPeriod: BlockNumber = 1 * DAYS;
 	pub const CooloffPeriod: BlockNumber = 7 * DAYS;
 	pub const MaxProposals: u32 = 100;
+	pub const ProposalFeeAmount: Balance = 10 * GRAINS_IN_LLM;
```
Comment (kacperzuk-neti) [inline bin/node/runtime/src/lib.rs:795]:
```suggestion
	pub const ProposalFeeAmount: Balance = 100 * DOLLARS;
```

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (kacperzuk-neti) [inline bin/node/runtime/src/lib.rs]:
```suggestion
	pub const LaunchPeriod: BlockNumber = 36 * DAYS;
```

File: frame/democracy/src/tests/public_proposals.rs [hunk 3]
```diff
@@ -146,9 +148,9 @@ fn blacklisting_should_work() {
...
 		System::set_block_number(0);
-		assert_ok!(propose_set_balance(1, 2, 2));
-		assert_ok!(propose_set_balance(1, 4, 4));
-		assert_ok!(propose_set_balance(1, 3, 3));
+		assert_ok!(propose_set_balance(3, 2, 2));
+		assert_ok!(propose_set_balance(3, 4, 4));
+		assert_ok!(propose_set_balance(3, 3, 3));
 		fast_forward_to(2);
 		assert_ok!(Democracy::vote(RuntimeOrigin::signed(1), 0, aye(1)));
 		fast_forward_to(4);
```
Comment (kacperzuk-neti) [inline frame/democracy/src/tests/public_proposals.rs:156]: Missing new test verifying that we actually charged the fee to the user (a.k.a. balance after propose should be equal to balance before - fee amount)

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

File: frame/democracy/src/tests/public_proposals.rs (hunk 3 above)
Comment (MichalLustykNeti) [reply frame/democracy/src/tests/public_proposals.rs:168]: I added "creating_proposal_takes_fee"

File: frame/democracy/src/tests/public_proposals.rs [hunk 4]
```diff
@@ -60,16 +60,30 @@ fn deposit_for_proposals_should_be_returned() {
...
+#[test]
+fn creating_proposal_takes_fee() {
 	new_test_ext().execute_with(|| {
-		assert_noop!(propose_set_balance(1, 2, 0), Error::<Test>::ValueLow);
+		System::set_block_number(0);
+		assert_eq!(Balances::free_balance(1), 10);
+
+		assert_ok!(propose_set_balance(1, 2, 5));
+
+		assert_eq!(Balances::free_balance(1), 0);
```
Comment (kacperzuk-neti) [inline frame/democracy/src/tests/public_proposals.rs:86]: `free_balance` could be reduced by `reserve` or `lock`. We want to actually take the coins, so we should check all balance.

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

**Review by kacperzuk-neti — APPROVED** [review]

**Comment by DorianSternVukotic** [comment]
I dont understand some namings and why we use slashing logic, maybe its best to leave it for a call

**Comment by DorianSternVukotic** [comment]
> > I dont understand some namings and why we use slashing logic, maybe its best to leave it for a call
> > At this PR slash is used only to get fee for creating new proposal from user it will slash only value that we specify at **ProposalFeeAmount** and the slashed many will go to **ProposalFee**
> 
> So for now **ProposalFeeAmount** is 100 * DOLLARS and **ProposalFee** is Treasury so user to create pr need to simply pay 100 to Treasury. This is not typical slash where we slash everything it is used here only to make user pay for his action.

why use slash instead of other methods for paying, if any ? Will it show up in slash history ?

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]
See https://github.com/liberland/liberland_substrate/pull/333#issuecomment-1803750422

**Review by kacperzuk-neti — APPROVED** [review]

**Pull request merged**